src/MongoDB.Driver.Core/Core/Servers/Server.cs

```diff
@@ -235,6 +241,20 @@ private void OnDescriptionChanged(object sender, ServerDescriptionChangedEventAr
             }
         }
 
+        private void OnMonitorDescriptionChanged(object sender, ServerDescriptionChangedEventArgs e)
+        {
+            var currentDescription = Interlocked.CompareExchange(ref _currentDescription, value: null, comparand: null);
+
+            // We cannot use FresherThan(e.NewServerDescription.TopologyVersion, currentDescription.TopologyVersion)
+            // because due to how// TopologyVersions comparisons are defined, IsStalerThanOrEqualTo(x, y) does not imply
+            // FresherThan(y, x)
```

Comment:
But could we use `IsFresherThanOrEqualTo`?

The code might read better: if the new description is fresher than or equal to the current description update the current description.